lib/Form/Checkbox/Group.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Checkbox from './';
+
+export default class Group extends Component {
+  static propTypes = {
+    choices: PropTypes.arrayOf(PropTypes.shape({
+
+    })).isRequired,
+    onChangeHandler: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const selected = props.choices.filter(c => c.checked);
+    this.state = { choices: props.choices, selected };
+    this.onChangeHandler = this.onChangeHandler.bind(this);
+    this.choiceIsAlreadySelected = this.choiceIsAlreadySelected.bind(this);
+    this.removeChoice = this.removeChoice.bind(this);
+    this.addChoice = this.addChoice.bind(this);
+  }
+
+  onChangeHandler(e) {
+    const choice = this.state.choices.find(c => c.id === e.target.id);
+
+    const selected = this.choiceIsAlreadySelected(choice) ? this.removeChoice(choice) : this.addChoice(choice);
```

Comment:
When we get `choice` don't we already get the checked/selected state? So...;

```
const choice = this.state.choices.filter(c => c.id === e.target.id);

if (choice.selected) ?
  this.removeChoice(choice) :
  this.addChoice(choice);
```

It's also worth noting that `find` isn't supported in IE. So I have changed that in my explain above.

Reply:
This is linked with my comment about `addChoice`. It'll show how we can edit the selected value of the choice within state and then use that in here.

Reply:
This isn't quite true, the `state.choices` will contain any choices that were initially checked when we first created the component, but if any choices are then checked/unchecked, we don't update the checked property on the choices, but add the whole choice to the `state.selected` array. Do you think we should worry about this inconsistency?

Reply:
We'll see when we give it a test. I'm not sure if it'll be an issue or not.